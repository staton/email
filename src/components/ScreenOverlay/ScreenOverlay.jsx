import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    isVisible: PropTypes.bool.isRequired,
    onClick: PropTypes.func,
    zIndex: PropTypes.number.isRequired
};

const defaultProps = {
    onClick: () => {}
};

export class ScreenOverlay extends Component {

	render() {
		return (
            <div 
                className={this.getClassNames()}
                onClick={this.props.onClick}
                style={{ zIndex: this.props.zIndex }}
            >
            </div>
        );
    }

    getClassNames() {
        const screenOverlayClassName = 'ScreenOverlay';
        const screenOverlayHiddenClassNames = screenOverlayClassName + ' ScreenOverlay--hidden';

        return (this.props.isVisible)
            ? screenOverlayClassName
            : screenOverlayHiddenClassNames;
    }

}

ScreenOverlay.propTypes = propTypes;
ScreenOverlay.defaultProps = defaultProps;

export default ScreenOverlay;