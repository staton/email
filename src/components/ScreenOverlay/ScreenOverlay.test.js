import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import ScreenOverlay from './ScreenOverlay';

describe('<ScreenOverlay />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(
            <ScreenOverlay 
                isVisible={true}
                zIndex={12}
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('has correct classes to handle visibility when it should be visible', () => {
        const wrapper = shallow(
            <ScreenOverlay
                isVisible={true}
                zIndex={12}
            />);
        expect(wrapper.find('.ScreenOverlay').length).toEqual(1);
        expect(wrapper.find('.ScreenOverlay--hidden').length).toEqual(0);
    });

    it('has correct classes to handle drawer visibility when it should be hidden', () => {
        const wrapper = shallow(
            <ScreenOverlay
                isVisible={false}
                zIndex={12}
            />);
        expect(wrapper.find('.ScreenOverlay').length).toEqual(1);
        expect(wrapper.find('.ScreenOverlay--hidden').length).toEqual(1);
    });

    it('calls onClick when clicked', () => {
        const onClick = jest.fn();
        const wrapper = mount(
            <ScreenOverlay
                isVisible={true}
                onClick={onClick}
                zIndex={12}
            />
        );
        wrapper.find('.ScreenOverlay').simulate('click');
        expect(onClick).toHaveBeenCalledTimes(1);
    });

});