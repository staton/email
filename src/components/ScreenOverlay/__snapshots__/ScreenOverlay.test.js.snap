// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenOverlay /> renders correctly 1`] = `
<div
  className="ScreenOverlay"
  onClick={[Function]}
  style={
    Object {
      "zIndex": 12,
    }
  }
/>
`;
